src/Components/Components/src/CascadingQueryValueProvider.cs

```diff
@@ -0,0 +1,224 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.AspNetCore.Components.Routing;
+using Microsoft.AspNetCore.Internal;
+
+namespace Microsoft.AspNetCore.Components;
+
+internal sealed class CascadingQueryValueProvider : IComponent, ICascadingValueSupplierFactory, IDisposable
```

Comment:
Great question, thanks for pointing this out! It turns out `SetParametersAsync` was in fact getting called once per cascading query parameter rather than once per component.

I've since updated my implementation to avoid this.

Reply:
Yeah, I am with Steve here, the updates to all query parameters should be "atomic".